behavior.nls

```diff
@@ -425,3 +394,183 @@ to move-around
     ]
   ]
 end
+
+
+
+
+
+; agents working on a not fully researched attacked argument will try to find a defense
+; for that attack, by staying on the current argument
+; if a child argument is discovered that can provide a defense, the 
+; agent moves there
+; once an argument is fully researched all its relations are discovered, then an agent
+; can move on and can try to find a defense in another branch, further away
+to find-defense
+  ask agents with [not communicating][
+    let myx xcor
+    let myy ycor
+    let myargu one-of turtles with [(breed = starts or breed = arguments) and
+      xcor = myx and ycor = myy]    
+    let curagent self
+    let myattacks []
+    let not-admissible []
+    
+    if admissible-subj-argu != 0 and not empty? admissible-subj-argu [
+      let info-not-admissible filter [not member? ? admissible-subj-argu] subjective-arguments
+      foreach info-not-admissible [
+        set not-admissible lput item 0 ? not-admissible
+      ]
+    ]
+    
+    if member? myargu not-admissible and [color] of myargu != red[
+      ; first find all attacking arguments from the same theory that attack
+      ; the attacking argument of the currently researched argument
+      ask myargu [
+        ask my-in-attacks [
+          ask myargu [
+            ask other-end [
+              if not any? in-attack-neighbors with [mytheory = [mytheory] of myargu][
+                set myattacks lput self myattacks
+              ]
+            ]
+          ]
+        ]
+      ]
+      
+      ; if one of these arguments is supported by the current argument
+      ; move to this argument
+      if not empty? myattacks [
+        ask turtles with [breed = arguments and in-support-neighbor? myargu and color != gray][
+          let nextargu self
+          let moved false
+          foreach myattacks [
+            let myattacker self
+            if in-attack-neighbor? nextargu [
+              ask curagent [
+                if not moved [
+                  move-to nextargu
+                  set moved true
+                ]
+              ]
+            ]
+          ]
+        ]
+      ]
+    ]
+  ]
+end
+
+
+
+
+
+; every five time steps the landscape is updated:
+; new arguments become visible and attacks are discovered
+to update-landscape
+  ask turtles with [breed = arguments or breed = starts][
+    let myx xcor
+    let myy ycor
+    ; discoveries only occur when an agent is working on that argument
+    ; working on an argument means that the agent did not communicate in that round
+    if any? turtles with [breed = agents and xcor = myx and ycor = myy and not communicating][
+      set researcher-ticks researcher-ticks + 1
+      ; for arguments that have still undiscovered relations/neighbors and are not fully researched
+      if ticks mod 5 = 0 and color != red [
```

Comment:
This code is part of the "update-landscape" procedure and determines whether new in-/out-attacks and/or child arguments of the respective argument are discovered.  Lets call what is happening here "discovery". For it to happen on a given argument the following criterion must be satisfied:

1.  The world must be in a round where mod 5 = 0 (i.e. round 0, 5, 10,...)
2. There must be at least one agent on the argument who is not communicating. (Agents might be communicating in mod 5 = 0 rounds (=ticks))
3. If 1. and 2. are satisfied there are more conditions to be fulfilled for the respective argument to actually discover a child argument and/or an in-out-attack which I will not mention here.

The change which took place here was to Point 1: mod 5 = 1  was changed to mod 5 = 0. I.e. the discovery can now only occur in rounds where agents are potentially communicating. This will cause the exploration to be blocked much more than before by communicating: 
The following assumes for visualization the extreme case that every agent is communicating every time they can i.e. every 5 rounds and compares it to the other extreme situation where no communication would happen:

- Before the change: Every 5 rounds one research tick for the argument is lost in comparison to the no communication case. I.e. instead of 25 rounds agents now need ~30 rounds to turn an argument from the first color ("lime") to the last ("red") 
- After the change: The above is true here as well, but in addition there will be no discovery of in-/out-attacks and/or child arguments before the argument is turned all the way to red. Note: Research ticks are added every round. That's why they're not fully blocked by the change but only every 5th round.

Iirc this 2nd discovery behavior ("after") is something we deliberately decided against. Is there any particular reason for reinstating this behavior?